tests/integ/modin/series/test_case_when.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from typing import List
+
+import modin.pandas as pd
+import pandas as native_pd
+import pytest
+from pandas.api.types import is_scalar
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from tests.integ.modin.sql_counter import SqlCounter, sql_count_checker
+from tests.integ.modin.utils import assert_series_equal, eval_snowpark_pandas_result
+
+
+def _verify_case_when(series: native_pd.Series, caselist: List[tuple]) -> None:
+    native_res = series.case_when(caselist)
+    caselist = [
+        (
+            pd.Series(cond) if isinstance(cond, native_pd.Series) else cond,
+            pd.Series(repl) if isinstance(repl, native_pd.Series) else repl,
+        )
+        for cond, repl in caselist
+    ]
+    snow_res = pd.Series(series).case_when(caselist)
+    assert_series_equal(snow_res, native_res)
+
+
+@pytest.mark.parametrize(
+    "repl", [native_pd.Series([11, 12, 13, 14]), [11, 12, 13, 14], 99]
+)
+@pytest.mark.parametrize(
+    "cond", [native_pd.Series([True, False, True, False]), [True, False, True, False]]
+)
+def test_case_when(cond, repl):
+    with SqlCounter(query_count=1, join_count=1 if is_scalar(repl) else 2):
+        series = native_pd.Series([1, 2, 3, 4])
+        _verify_case_when(series, [(cond, repl)])
+
+
+@sql_count_checker(query_count=1, join_count=1)
+def test_case_when_misaligned_index():
+    series = native_pd.Series([1, 2, 3, 4, 5, 6])
+    cond = native_pd.Series([True, False, True, False, True], index=[0, 1, 2, 6, 7])
+    _verify_case_when(series, [(cond, 99)])
+
+
+@sql_count_checker(query_count=1, join_count=2)
+def test_case_when_mulitple_cases():
+    series = native_pd.Series([1, 2, 3, 4, 5, 6])
+    cond1 = native_pd.Series([True, False, True, False, True])
+    cond2 = native_pd.Series([False, True, False, False, True])
+    caselist = [(cond1, 98), (cond2, 99)]
+    _verify_case_when(series, caselist)
+
+
+@pytest.mark.parametrize("caselist", [[], [()], [(97, 98, 99)]])
+@sql_count_checker(query_count=0)
+def test_case_when_invalid_caselist(caselist):
```

Comment:
Can we also add tests for the TypeErrors raised as well as not implemented for callable?